### PR TITLE
fix xdecode and some of the spdemo/scaldemo tests

### PR DESCRIPTION
**Pull request by othiman**
- xencode-verify still fails
- due to https://github.com/openitu/STL/blob/master/src/utl/makefile.unx spdemo8-2-verify, spdemo10-1-verify, spdemo14-1-verify and spdemo15-1-verify are OK to fail, how can we componsate this?
- invert endianess for src/is54/test_data/voice.src to fix scaldemo2 test

### CODE REVIEW

File: src/unsup/CMakeLists.txt [hunk 1]
```diff
@@ -80,7 +80,7 @@ add_test(fdelay5-verify ${CMAKE_COMMAND} -E compare_files test_data/delay-15.tst
 add_test(fdelay6 ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/fdelay -hex 0xF test_data/litend.src test_data/delay-0f.tst)
 add_test(fdelay6-verify ${CMAKE_COMMAND} -E compare_files test_data/delay-0f.tst test_data/delay-15.ref)
 
-add_test(NAME xdecode1 COMMAND ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/xdecode voicet.uue WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/test_data)
+add_test(NAME xdecode1 COMMAND ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/xdecode voice.uue WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/test_data)
```
Comment (dennisguse) [inline src/unsup/CMakeLists.txt]: Could you move this part into a separate commit or even pull requests?
It is easier to track this way and the commit log is cleaner.

**Comment by othiman** [comment]
The fix for scaldemo2 depends on PR #88.

**Comment by othiman** [comment]
needed to rebase the PR caused by the force revert push

File: src/utl/CMakeLists.txt [hunk 2]
```diff
@@ -50,8 +50,8 @@ add_test(spdemo11-verify ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/cf -q test_data/sptst
 
 add_test(spdemo12 ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/spdemo -q -r 16 -right sp  test_data/sptst-r.s16 test_data/sptst-r.p16 100)
 add_test(spdemo13 ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/spdemo -q -r 16 -left  sp  test_data/sptst-l.s16 test_data/sptst-l.p16 100)
-add_test(spdemo13-verify ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/cf -q test_data/sptst-l.p16 test_data/sptst-r.p16)
-add_test(spdemo13-verify ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/cf -q test_data/sptst-l.p16 test_data/sptst-s.p16)
+add_test(spdemo13-1-verify ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/cf -q test_data/sptst-l.p16 test_data/sptst-r.p16)
+add_test(spdemo13-1-verify ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/cf -q test_data/sptst-l.p16 test_data/sptst-s.p16)
```
Comment (othiman) [inline src/utl/CMakeLists.txt]: hmm. I missed that one... However shouldn't the second one of the spdemo13-1-verify be spdemo13-**2**-verify? I think otherwise only the last test with the same name is executed.

**Comment by dennisguse** [comment]
Will fix it and merge tomorrow morning.

**Comment by ludomal** [comment]
While reviewing #107, I checked out the latest code and spdemo tests fails on my machine (osx) as the executable crashes. I am not sure why it passes on Linux; is Travis enabled for osx? I could not find the build.

Doing a bit of debugging, the exe crashes at line 538 `FIND_PAR_S (9, "_Left of right adjusted? ..... ", just, just);`. This line expects the 9th argument to be the justification parameter, but the tests pass 8 arguments only.

Section 530 to 538 should be reviewed as it looks like the parameters are incorrectly parsed.
